starfish/image/_filter/linear_unmixing.py

```diff
@@ -0,0 +1,107 @@
+from functools import partial
+from typing import Optional
+
+import numpy as np
+
+from starfish.imagestack.imagestack import ImageStack
+from starfish.types import Axes
+from starfish.util import click
+from ._base import FilterAlgorithmBase
+from .util import preserve_float_range
+
+
+class LinearUnmixing(FilterAlgorithmBase):
+
+    def __init__(self, coeff_mat: np.ndarray) -> None:
+        """Image scaling filter
+
+        Parameters
+        ----------
+        coeff_mat : np.ndarray
+            matrix of the linear unmixing coefficients. Should take the form:
+            B = AX, where B are the unmixed values, A is coeff_mat and X are
+            the observed values.
+
+        """
+        self.coeff_mat = coeff_mat
+
+    _DEFAULT_TESTING_PARAMETERS = {"coeff_mat": np.array([[1, -0.25], [-0.25, 1]])}
+
+    @staticmethod
+    def _unmix(image: np.ndarray, coeff_mat: np.ndarray) -> np.ndarray:
+        """Clip values of img below and above percentiles p_min and p_max
+
+        Parameters
+        ----------
+        image : np.ndarray
+            image to be scaled
+
+        coeff_mat : np.ndarray
+            matrix of the linear unmixing coefficients. Should take the form:
+            B = AX, where B are the unmixed values, A is coeff_mat and X are
+            the observed values.
```

Comment:
This is the case, right? 
```suggestion
            the observed values. coeff_mat has shape (n_ch, n_ch).
```